### ISSUE TITLE
Wrong credidentials throw a ValueError

### ISSUE THREAD

**Issue by boltgolt** [costastf/toonapilib#1]
In the old toonlib, a more descriptive InvalidCredentials error would be thrown. Is this by design?

**Comment by costastf** [comment]
The code should raise either ChallengeCodeError or InvalidCredentials. The autentication is Oath2 and is a bit more involved with the backend service not being very verbose on errors. If you are getting a ValueError though, something is wrong. If you could paste a debug trace I could look into it.

**Comment by boltgolt** [comment]
It seems like it catches email/password errors, but not problems with the key or secret.

```
Traceback (most recent call last):
  File "/home/user/.homeassistant/deps/lib/python3.5/site-packages/toonapilib/toonapilib.py", line 93, in __init__
    self._authenticate()
  File "/home/user/.homeassistant/deps/lib/python3.5/site-packages/toonapilib/toonapilib.py", line 119, in _authenticate
    self._token = self._get_token(code)
  File "/home/user/.homeassistant/deps/lib/python3.5/site-packages/toonapilib/toonapilib.py", line 150, in _get_token
    return self._retrieve_token(payload)
  File "/home/user/.homeassistant/deps/lib/python3.5/site-packages/toonapilib/toonapilib.py", line 169, in _retrieve_token
    'Got: {}'.format(response.json()))
ValueError: Incomplete token response received. Got: {'fault': {'faultstring': 'Invalid client identifier {0}', 'detail': {'errorcode': 'oauth.v2.InvalidClientIdentifier'}}}
```

**Comment by costastf** [comment]
Well, it was by design actually, so i was actually throwing a ValueError because I was to lazy to try out all the errors. Anyway, that is fixed and I have pushed version 1.0.0 with the appropriate exception per issue. So it throws InvalidCredentials on username/password issue InvalidConsumerKey and InvalidConsumerSecret. Thanks for reporting!
